enterprise/coderd/userauth.go

```diff
@@ -33,43 +35,49 @@ func (api *API) setUserGroups(ctx context.Context, logger slog.Logger, db databa
 		}
 
 		// Delete all groups the user belongs to.
-		err = tx.DeleteGroupMembersByOrgAndUser(ctx, database.DeleteGroupMembersByOrgAndUserParams{
-			UserID:         userID,
-			OrganizationID: orgs[0].ID,
-		})
+		// nolint:gocritic // Requires system context to remove user from all groups.
+		err = tx.RemoveUserFromAllGroups(dbauthz.AsSystemRestricted(ctx), userID)
 		if err != nil {
 			return xerrors.Errorf("delete user groups: %w", err)
 		}
 
-		if createMissingGroups {
-			// This is the system creating these additional groups, so we use the system restricted context.
-			// nolint:gocritic
-			created, err := tx.InsertMissingGroups(dbauthz.AsSystemRestricted(ctx), database.InsertMissingGroupsParams{
-				OrganizationID: orgs[0].ID,
+		// TODO: This could likely be improved by making these single queries.
+		// 	Either by batching or some other means. This for loop could be really
+		//	inefficient if there are a lot of organizations. There was deployments
+		//	on v1 with >100 orgs.
```

Comment:
This is just phase 1 to getting the codebase to accept multi orgs. I do not want to prematurely optimize by coming up with some batching that might be removed.

I think we should move to `pgx` as our pg driver in sqlc to get batching support.